code/io/keycontrol.cpp

```diff
@@ -490,11 +490,11 @@ void debug_cycle_player_ship(int delta)
 	ship_info	*sip;
 	while ( TRUE ) {
 		si_index += delta;
-		if ( si_index > Num_ship_classes ){
+		if ( si_index >= static_cast<int>(Ship_info.size()) ){
```

Comment:
Well, if `si_index == Ship_info.size()`, then the index access on line 499 below (`sip = &Ship_info[si_index];`) will be out-of-bounds.